### ISSUE TITLE
Create new community conversations system.

### ISSUE THREAD

**Issue by halkeye** [nodeschool/vancouver#80]
As much as I like discourse, https://community.vancouvertech.com didn't really take off the way it should.

I don't mind doing the bulk of monitoring/moderation/etc but I'd like something we actually have some control over and is safe and welcoming like our actual events.

My preferences are:
1) New slack, nodeschoolyvr, or even nodeschool/#yvr (if @Qard or someone has contact with top level nodeschool people)
2) https://spectrum.chat/
3) Vancouvertech
4) Existing yvrdev slack

I'd like to make a formal decision soon though, so we can update our meetups/emails/whatever else.

**Comment by halkeye** [comment]
I was looking at the nodeeschool wikis and a lot of them have their own slacks.

**Comment by Qard** [comment]
We could use a channel in wealljs, they're much better about moderation than yvrdev. Personally, I would try to avoid creating a brand new slack where we'd have to fully moderate on our own.

No idea about how the get set up in `nodeschool/#yvr`, but that might be worth looking into. Probably some details in the nodeschool admin repos somewhere. 🤔

**Comment by keywordnew** [comment]
I'm +1 to using weAllJS, mainly for moderation, and the fact that it's JS. There's also Node Slackers.

My main preference is to not use another medium or Slack that people don't already use, unless there are clear advantages that suggest it can overtake Slack elsewhere at some point. The above two Slacks are likely to be already used by people using Node.js

**Comment by halkeye** [comment]
so I'm -1 to using someone elses system. +1 to our own

> where we'd have to fully moderate on our own.

I don't really understand the hesitance to moderate. Has it been a problem before? Why are we expecting someone else to do work we don't want to do ourselves? We currently have 300 people in meetup, i don't expect that many on slack in the slightest

Node school has really been about getting people to safely and comfortably work together. Having random strangers from the internet to me seriously takes away from that.

We would also gain the ability to have any integrations we wanted, meetup annoucements, maybe a slack bot workshop, i don't know.

The global nodeschool slack outweighed that to me because it let people have a chance to find other chapters if they are traveling or whatever. (look at https://github.com/rladies/starter-kit/wiki/Tech-Infrastructure-for-Chapters for an example of that)

**Comment by halkeye** [comment]
also looking at https://github.com/nodeschool/organizers I think pinging @gyaresu is worth it, so I'm doing so....ping

**Comment by halkeye** [comment]
oh and I'll throw out why I like discourse (or even github issues) instead of slack. Slack is very conversational. Any issues will get lost after an hour or so, meaning its very hard to follow up.

**Comment by keywordnew** [comment]
I hear you, and I support the need for being able to not have random strangers from the internet.

I also don't like Slack for its being conversational. I find myself not being on even the Slacks I've already joined for that reason.

I loved discourse and I wish it had taken off! And, now that it has slowed, I don't think a platform that only serves to connect for NodeSchoolYVR will have traction. The global nodeschool slack at least will have more activity because of the broad base. But this is just my opinion. I'd love to hear more about what y'all think.

Based on the conversation here, I don't feel there's a clear consensus. How do y'all feel about getting on a zoom to seek a consensus real-time?

**Comment by halkeye** [comment]
I'm on vacation right now, so i'd like to discuss more by github for a bit.

I think my vote is spectrum for now, based on my original comments, since its free, hosted, and threaded. But i'd like to hear more reasonings/ideas/cons/etc from more people.

Moderation, I'm pretty much always around, and hopefully others will start being around more too so i can volunteer, but in general I prefer to be hands off till something comes up.

**Comment by Qard** [comment]
I don’t feel strongly any particular way. If you think we should have our own Slack, I’m fine with that. I just personally try to stick to established communities with a big enough user base to have moderation available at any time. I’ve seen Slack communities turn into a trash-fire when the mods were asleep. 🙄

Also, ETOOMANYSLACKS.

**Comment by keywordnew** [comment]
My opinion here is that a communication platform that serves only NodeSchoolYVR will go the way of Gitter. Too few users to convince people that it's worth keeping the app around, lower engagement, then too little activity, so fewer users will join, and so on.

A useful communication platform is one where people find value from actually being on it. And after the experience with Gitter and Discourse, to me that means being on a platform that serves multiple communities connected by some common thread: be it JS, NodeSchool, Node.js etc.

It would help to have a priority list of what NodeSchoolers want from this platform. For example, looks like we want to have a healthy environment (from the moderation discussion), we want it to persist with context (from the comment about Slack being too conversational), and we want it to not exist in isolation (from the comments about using inactive instances).

How can we make it so all these are satisfied?

**Comment by halkeye** [comment]
> Too few users to convince people that it's worth keeping the app around

So to me notifications are important, gitter didn't notify you properly so messages were missed all the time. Also we need to start agressively marketing it, mention it every event, put it up on our "Signs" and stuff.

So lets not focus on what yet, lets do as manil's been saying and figure out more requirements.

I'm going to create a spectrum chat for myself to see what features it has.

![image](https://user-images.githubusercontent.com/110087/47127666-01d24780-d243-11e8-83fb-e43f0506e034.png)

So need to decide if we want it public (anyone can join, low effort, higher moderation) or private (invite only, maybe tied to meetup api or a QR code at events or something).

![image](https://user-images.githubusercontent.com/110087/47127708-24646080-d243-11e8-88eb-8bb52d1fa869.png)

An argument could be made for ~spectrum~ public form for everyone, and slack for "organization", but since nodeschool charter says organization should be pretty public, for now I'd rather use github issues, or zoom/slack(yvrdev) for the few leftovers.

Spectrum has multiple channels, but I don't think its all that needed if things are threaded.

I'd like to avoid Private Messaging support, since I think that'll, at least short term, lead to harassment, creepy messages, etc.

**Comment by halkeye** [comment]
Come play with my spectrum https://spectrum.chat/gavin-mogan/join/GEVAiLnKN

I looked into discourse, but cheapest option is hosting it yourself on DO (or my home server), but thats like $20/month, so unless we think agressively advertising and adding some sort of auto import to vancouver tech, i think we can eliminate that one.

**Comment by keywordnew** [comment]
At the meetup today there was discussion about communication conversation channels. 

I heard compelling feedback about the existing Slack instance, in line with the discussion about its community health so far. There was a clear want for a space where people in it knew each other and felt safe to ask about stuff.

I have some experience with Slack. I don't have much familiarity with Spectrum. I initialized a test Spectrum.chat space, and invited just the organizers and people who were present to join. 

I'm sorry for this unilateral action, which I realize in retrospect is presumptuous. It wasn't meant to be a decision related to this #80. Its intent was to be a test that included community members beyond the regular organizers to see if this platform would satisfy the ask for a virtual space serving one community which had the capacity to be less ephemeral than Slack.

**Comment by Fishrock123** [comment]
A custom Slack has worked out pretty good for NodeSchool Toronto, for what it’s worth.

**Comment by halkeye** [comment]
I'm going to have to make a pro/cons list soon of spectrum, slack, and shared slack (any other ideas so far?)

https://spectrum.chat/thread/439a673b-7808-4bf1-b414-a3c869838d92

Right now they don't have an API to invite users or make posts, which would have been a nice to have, but not all that bad.

**Comment by halkeye** [comment]
> A custom Slack has worked out pretty good for NodeSchool Toronto, for what it’s worth.

yea we just don't want to jump into that decision, so we want to discuss and research and stuff.

**Comment by halkeye** [comment]
Manil and I have been poking around the new spectrum he setup (ping him or I if you want beta access).

https://gist.github.com/halkeye/15282216a3e280ea09e466c3bff42a53 is all the notes I have so far. I was sick so I didn't get very far researching last week, but hope to finish it off this week.

**Comment by keywordnew** [comment]
^^ That Gist is detailed 💯

Good idea to optionally guide new open source contributors to contribute on Spectrum!

**Comment by halkeye** [comment]
@chowdhurian @Qard anything else you want me to look into?

My vote so far is still spectrum

**Comment by halkeye** [comment]
Can we lock down a time to finalize this so a we can start for the new year? Maybe a quick chat at Cascadia or zoom after that.

**Comment by keywordnew** [comment]
Am down to go on a Zoom! @Qard is in UTC+7. Really early morning seems Pacific-time seems likely more convenient.

**Comment by halkeye** [comment]
My vote is still spectrum. 

I'm pretty flexible until monday morning until next weekend.

**Comment by keywordnew** [comment]
I'm in for Spectrum too.

Circling back around to this so we can get on a call, discuss, and resolve!

Else maybe we can resolve it right here?

**Comment by halkeye** [comment]
I vote before or after the meeting on the 12th

**Comment by keywordnew** [comment]
^^ 👍

**Comment by keywordnew** [comment]
@emclaren @halkeye @Qard after a bunch of conversations among us after the NodeSchool session at Clio yesterday, a possible next step is to just support conversations through meetup.com

There are reservations about the ability moderate on meetup.com

**Comment by Qard** [comment]
I'm good with trying it, if someone wants to do some testing to see if the tools and structure seem reasonable enough. Moderation is important. Notifications controls would be good too.

**Comment by halkeye** [comment]
@Qard I'm so indifferent at this point. I would say make an executive decision as lead organizer, and we'll stick with it the rest of the year and revisit again at the end.

**Comment by Qard** [comment]
How about we just stick to meetup.com discussions for now then? If we run into any issues, we can just bail out and turn off discussions access at any time.

**Comment by halkeye** [comment]
I'm good with that.

I will say that meetup notification system is a nightmare. Because I had the mobile app installed a couple phones ago, they disabled email notifications. I tried opening a ticket and got no response. Had to raise a fuss on twitter (a month later) before I got it fixed.

So maybe get a couple discussions going and see if any of our regulars contribute / ask if it's seen at the next event?
